### ISSUE TITLE
Transition to different animation

### ISSUE THREAD

**Issue by daimahou** [KybernetikGames/animancer#143]
Hello, maybe I didnt understand how to properly use the feature but I ran into a null pointer exception. 

Here is the scenario : 

I want to create a melee combo system.

1. When I trigger the attack input, the combo start the first animation. 

Animation are played with the following code : 

`

            var animation = _weapon.GetComboAnimation(_currentCombo, _comboIndex);

            _state = character.animancer.Play(animation);
            _state.Events.OnEnd = endAttack;
            _shouldPerformCombo = false;

`

2 The animation has an event "combo" set up which triggers a function in my code. If the events goes off while the shouldPerformCombo input buffer is true, then the next animation is played using the same code.

3. Animation goes off properly and there is no sign of problems in the game but I do get a concerning null pointer in the console. Here is the stack trace.

`

          Exception] NullReferenceException: Object reference not set to an instance of an object
          AnimancerState+EventDispatcher.NextEvent() at /Plugins/Animancer/Internal/Core/AnimancerState.EventDispatcher.cs:641
          
          AnimancerState+EventDispatcher.CheckGeneralEvents() at 
          /Plugins/Animancer/Internal/Core/AnimancerState.EventDispatcher.cs:528
          
          Animancer.AnimancerState+EventDispatcher.Animancer.IUpdatable.Update() at 
          /Plugins/Animancer/Internal/Core/AnimancerState.EventDispatcher.cs:265
          
          AnimancerPlayable.UpdateAll() at /Plugins/Animancer/Internal/Core/AnimancerPlayable.cs:1194
          
          Debug.LogException()
          
          AnimancerPlayable.UpdateAll() at /Plugins/Animancer/Internal/Core/AnimancerPlayable.cs:1199
          
          Animancer.PostUpdate.PrepareFrame() at /Plugins/Animancer/Internal/Core/AnimancerPlayable.cs:1293

`

---

I don't really understands what is the problem here, I guess animancer does not like the way I trigger the following animations of the combo. Maybe the fact the end event does not get triggered is problematic ? If call my combo in the end event, I dont get any issues.

Can you please help me resolve this problem ?

Best regards,

**Comment by daimahou** [comment]
I wanted to clarify that I want my combo to work this way because the animation can chain in a combo, but if the combo doesnt trigger on time, then I want to play the recovery bit of the animation while keeping my player locked in the animation.

**Comment by KybernetikGames** [comment]
Yeah, the issue is that regular events aren't expecting you to play something else where end events do. Playing something else clears the events before checking if the next event also needs to be triggered.

Just change the:
```
                if (_NextEventIndex == RecalculateEventIndex)
                    return false;
```
To:
```
                if (_NextEventIndex == RecalculateEventIndex ||
                    _Events == null)
                    return false;
```
I haven't tested it yet, but that should fix it.

**Comment by daimahou** [comment]
That fixed it thank you.

**Comment by KybernetikGames** [comment]
I'll keep the issue open until the fix is actually in the released version.

**Comment by KybernetikGames** [comment]
This issue is fixed in [Animancer v7.2](https://kybernetik.com.au/animancer/docs/changes/animancer-v7-2).
